src/redis/driver_impl.go

```diff
@@ -103,6 +94,29 @@ func NewAuthTLSPoolImpl(scope stats.Scope, auth string, url string, poolSize int
 		stats: newPoolStats(scope)}
 }
 
+func NewAuthPoolImpl(scope stats.Scope, auth string, url string, poolSize int) Pool {
```

Comment:
This is basically the same function as the one above. Can you just merge it all into a single creation function that decides whether to use TLS, auth, etc. independently?

Reply:
done